csm_web/scheduler/views/student.py

```diff
@@ -91,3 +93,40 @@ def attendances(self, request, pk=None):
             f"<Attendance:Failure> Could not record attendance for User {log_str(request.user)}, errors: {serializer.errors}"
         )
         return Response(serializer.errors, status=status.HTTP_422_UNPROCESSABLE_ENTITY)
+
+    @action(detail=True, methods=["put"])
+    def submit_attendance(self, request, pk=None):
+        """
+        Attempts to change the attendance object associated with the given student.
+
+        Format of request.
+            request.data is a dictionary with
+                'attempt': string of attempt for word of the day    
+                'section_occurrence': pk for section occurrence for that week
+        """
+        request_time = timezone.now()
+
+        student = get_object_or_error(self.get_queryset(), pk=pk)
+
+        # Internal server error
+        if student.user != request.user:
+            return Response(status=status.HTTP_404_BAD_REQUEST)
+
+        section_occurrence = SectionOccurrence.objects.filter(pk=request.data['section_occurrence']).first()
+
+        section_occurrence_time_limit = datetime.datetime.combine(section_occurrence.date, datetime.datetime.min.time(), tzinfo=timezone.now().tzinfo) + datetime.timedelta(days=1)
+        
+        # Reject any requests made after the deadline (midnight the next day)
+        if request_time > section_occurrence_time_limit:
+            return Response(status=status.HTTP_403_FORBIDDEN)
+
+        # If the attempt at the word of the day is incorrect, deny request
+        if section_occurrence.word_of_the_day.lower() != request.data['attempt'].lower():
+            return Response(status=status.HTTP_403_FORBIDDEN)
```

Comment:
Add messages to these responses explaining what happened (ie "deadline passed" or "incorrect word of the day"). This way, the frontend knows what to tell the user.

Reply:
Sorry, can you clarify how I should add the messages? Based on the django rest fw documentation, I think the only argument that makes sense would be data.